test/controllers/concerns/active_insights/basic_authentication_test.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class ActiveInsights::BasicAuthenticationTest < ActionDispatch::IntegrationTest
+
+  class TestController < ActionController::Base
+    include ActiveInsights::BasicAuthentication
+
+    def index
+      render plain: "Hello from TestController"
+    end
+  end
+
+  setup do
+    @controller = TestController.new
+    @original_routes = Rails.application.routes
+
+    Rails.application.routes.draw do
```

Comment:
We should be using with_routing here. This wont work on rails 8 with the new lazy route loading. 

https://edgeapi.rubyonrails.org/classes/ActionDispatch/Assertions/RoutingAssertions.html#method-i-with_routing

Reply:
Actually looking at the mission control pr, we should be able to test without having to setup a dummy controller and route. https://github.com/rails/mission_control-jobs/pull/214/files#diff-60419e9ff1eee7716008b46a404d4cd081cdd9186775d62354a7d33553952031

We should be able to setup the config and then hit the insights routes